src/ansys/fluent/core/launcher/launcher_arguments.py

```diff
@@ -0,0 +1,346 @@
+from enum import Enum
+import os
+from pathlib import Path
+import platform
+import subprocess
+import tempfile
+from typing import Any, Dict, Optional, Union
+
+from ansys.fluent.core.exceptions import DisallowedValuesError
+from ansys.fluent.core.fluent_connection import FluentConnection, PortNotProvided
+from ansys.fluent.core.launcher.custom_exceptions import (
+    DockerContainerLaunchNotSupported,
+    InvalidPassword,
+    IpPortNotProvided,
+    UnexpectedKeywordArgument,
+)
+from ansys.fluent.core.launcher.launcher_utils import check_docker_support, logger
+from ansys.fluent.core.session import _parse_server_info_file
+from ansys.fluent.core.session_meshing import Meshing
+from ansys.fluent.core.session_pure_meshing import PureMeshing
+from ansys.fluent.core.session_solver import Solver
+from ansys.fluent.core.session_solver_icing import SolverIcing
+from ansys.fluent.core.utils.fluent_version import FluentVersion
+from ansys.fluent.core.utils.networking import find_remoting_ip
+import ansys.platform.instancemanagement as pypim
+
+
+class LaunchMode(Enum):
+    """An enumeration over supported Fluent launch modes."""
+
+    STANDALONE = 1
+    PIM = 2
+    CONTAINER = 3
+    SLURM = 4
+
+
+class FluentMode(Enum):
+    """An enumeration over supported Fluent modes."""
+
+    MESHING_MODE = (Meshing, "meshing")
+    PURE_MESHING_MODE = (PureMeshing, "pure-meshing")
+    SOLVER = (Solver, "solver")
+    SOLVER_ICING = (SolverIcing, "solver-icing")
+
+    @staticmethod
+    def get_mode(mode: str) -> "FluentMode":
+        """Returns the FluentMode based on the provided mode string.
+
+        Parameters
+        ----------
+        mode : str
+            mode
+
+        Returns
+        -------
+        FluentMode
+            Fluent mode
+
+        Raises
+        ------
+        DisallowedValuesError
+            If an unknown mode is passed.
+        """
+        allowed_modes = []
+        for m in FluentMode:
+            allowed_modes.append(m.value[1])
+            if mode == m.value[1]:
+                return m
+        raise DisallowedValuesError("mode", mode, allowed_modes)
+
+    @staticmethod
+    def is_meshing(mode: "FluentMode") -> bool:
+        """Returns whether the current mode is meshing.
+
+        Parameters
+        ----------
+        mode : FluentMode
+            mode
+
+        Returns
+        -------
+        True if mode is FluentMode.MESHING_MODE or FluentMode.PURE_MESHING_MODE else False
+            bool
+        """
+        return mode in [FluentMode.MESHING_MODE, FluentMode.PURE_MESHING_MODE]
+
+
+def get_fluent_exe_path(**launch_argvals) -> Path:
+    """Get Fluent executable path. The path is searched in the following order.
+
+    1. ``product_version`` parameter passed with ``launch_fluent``.
+    2. The latest ANSYS version from ``AWP_ROOTnnn``` environment variables.
+
+    Returns
+    -------
+    Path
+        Fluent executable path
```

Comment:
```suggestion
    """Get the path for the Fluent executable file. The search for the path is performed in this order.

    1. Look at the ``product_version`` parameter passed with the ``launch_fluent()`` method.
    2. Look at the latest Ansys version in ``AWP_ROOTnnn``` environment variables.

    Returns
    -------
    Path
        Fluent executable path
```
Ansys is always lowercase except for in trademark statements.